### ISSUE TITLE
Jupyter Notebook comments/suggestions

### ISSUE THREAD

**Issue by CBurge95** [DHARPA-Project/kiara#39]
- Can downloading files be swapped for uploading/onboarding data? Can still provide the sample data used here as a file in a folder (thinking in terms of a workshop/tutorial) but also offers framework for users to upload their own material.
- Is it possible for onboarding/tabular/other plugins necessary to all modules be included automatically in the installation, rather than something manual/separate (would then include details in documentation for using operations but point to the fact that this has already been included)
- A general question: is there a way to prioritise or show python querying first? I think given this is primarily python at the moment, it would be helpful to start with python query and then suggest that it can be also be done in SQL. Can this include a more detailed explanation of how we convert this into a python object? [Cell Input 25] I think otherwise we also need to include instructions for SQL query/links to courses (assuming that these are basic coders who must have some python knowledge to start with) (is this what is meant by the To-do note in the final comment?)

**Comment by makkus** [comment]
- uploading data to a jupyter notebook is a bit harder, it really depends how the notebook is ran, and where. I opted for the download, because that way I can make sure the notebook will always work, independent of whether it's run locally, or on colab, etc. It's possible, but we'd have to define first who it should work for, and in which scenarios exactly. 

- it's possible to just install all available plugins, you just don't provide any plugin names in to the 'ensure' method

- this is a much bigger question, and we'd have to decide what exactly we want to teach people. If we go for the 'manual' Python stuff, then the question is why people wouldn't just use plain jupyter with a bit of pandas

The notebook I created was really only an example for how to use kiara within Python, and should really be ignored in terms of its content. That content should be created by the research team, with a tight focus on research practices, not what's possible in kiara. If we only want to have a kiara-python tutorial, we can do that too, of course, but I'm not sure how much value would be in that. 

I guess the first thing we have to do is define exactly what the purpose of the notebook should be, and who exactly is the audience. Again, my example notebook was not concerned with any of that, its a bare-bones show-case, and I realize now it's probably more a distraction than it helps because it sorta 'primes' people who look at it. So, maybe it'd be best to just start from scratch, once we have the definition of purpose and audience.
